### DISCUSSION TITLE
Device stack with Pico-PIO-USB?

### DISCUSSION THREAD

**Discussion by asit-fdraschbacher** [adafruit/Adafruit_TinyUSB_Arduino#347] [Q&A]
Hi,
I've been studying the code here, the Pico-PIO-USB project and TinyUSB sources over the last few days to figure out if Adafruit_TinyUSB_Arduino supports Pico-PIO-USB for devices. It seems like there is some code for it in https://github.com/adafruit/Adafruit_TinyUSB_Arduino/blob/master/src/portable/raspberrypi/pio_usb/dcd_pio_usb.c, but then again `CFG_TUD_RPI_PIO_USB` doesn't seem to be enabled anywhere.

Is this library supposed to support emulating USB devices through Pico-PIO-USB?

**Comment by hathach** [comment]
no, this arduino variant mostly use pio-usb for host.

**Comment by asit-fdraschbacher** [comment]
Does TinyUSB itself (not this Arduino library) support Pico-PIO-USB for devices?
